addons/binding/org.openhab.binding.evohome/src/main/java/org/openhab/binding/evohome/internal/api/ApiAccess.java

```diff
@@ -0,0 +1,198 @@
+                }
+            }
+
+            if (requestData != null) {
+                request.content(new StringContentProvider(requestData), contentType);
+            }
+
+            ContentResponse response = request.send();
+
+            logger.debug("Response: {}", response);
+            logger.trace("\n{}\n{}", response.getHeaders(), response.getContentAsString());
+
+            if ((response.getStatus() == HttpStatus.OK_200) || (response.getStatus() == HttpStatus.ACCEPTED_202)) {
+                String reply = response.getContentAsString();
+
+                if (outClass != null && out != null) {
+                    retVal = new Gson().fromJson(reply, outClass);
+                }
+            } else {
+                retVal = null;
+            }
+        } catch (TimeoutException | InterruptedException | ExecutionException e) {
+            logger.error("Error in handling request", e);
+        }
+
+        return retVal;
+    }
+
+    /**
+     * Issues an HTTP GET request on the API's URL, using an object that is serialized to JSON as input.
+     * Makes sure that the request is correctly formatted.*
+     *
+     * @param url The URL to query
+     * @param outClass The type of the requested result
+     * @param out An instance of the return type, ignored when value is null
+     * @return The result of the request or null
+     */
+    public <TOut> TOut doAuthenticatedGet(String url, Class<TOut> outClass, TOut out) {
+        return doAuthenticatedRequest(HttpMethod.GET, url, null, outClass, out);
+    }
+
+    /**
+     * Issues an HTTP request on the API's URL, using an object that is serialized to JSON as input.
+     * Makes sure that the request is correctly formatted.*
+     *
+     * @param url The URL to query
+     * @param requestContainer The object to use as JSON data for the request
+     */
+    public void doAuthenticatedPut(String url, Object requestContainer) {
+        doAuthenticatedRequest(HttpMethod.PUT, url, requestContainer, null, null);
+    }
+
+    /**
+     * Issues an HTTP request on the API's URL, using an object that is serialized to JSON as input.
+     * Makes sure that the request is correctly formatted.*
+     *
+     * @param method The HTTP method to use (POST, GET, ...)
+     * @param url The URL to query
+     * @param headers The optional additional headers to apply, can be null
+     * @param requestContainer The object to use as JSON data for the request
+     * @param outClass The type of the requested result
+     * @param out An instance of the return type, ignored when value is null
+     * @return The result of the request or null
+     */
+    private <TOut> TOut doRequest(HttpMethod method, String url, Map<String, String> headers, Object requestContainer,
+            Class<TOut> outClass, TOut out) {
+
+        String json = null;
+        if (requestContainer != null) {
+            Gson gson = new GsonBuilder().create();
```

Comment:
This is an expensive operation. You should keep the Gson instance as a field in the class and reuse it.